### ISSUE TITLE
add "tested by" feature to docs

### ISSUE THREAD

**Issue by aphearin** [astropy/halotools#274]
A significant portion of the documentation is dedicated to the test suite, and functions/classes frequently provide hyperlinks to the appropriate test suite docstring. This might be more cleanly handled if there were a "tested by" colored box in the docs in exact analogy to the "See also" colored box, as suggested by @yyao.

**Comment by eteq** [comment]
Hmm, this is a tricky thing because it requires interfacing ``py.test`` and sphinx.  But I like the idea!

There are a few ways to do this that I can imagine, but I think the easiest would be to try to use the output of the ``coverage`` pytest plugin. That might require some extra care in how the docs are generated though, particularly handling the case where ``coverage`` isn't present.

@aphearin - what do you think is the best way to handle the situation of tests failing/coverage not being present/whatever?  Should the sphinx build just proceed and not provide the boxes if there's a problem, or actually fail?

**Comment by aphearin** [comment]
I'm not sure we're talking about the same thing, @eteq . Let me try again. 

When writing a docstring, you know how you can write a subsection called "See also"? And then there is a specially marked section that appears? See, for example, the halo_mass_to_halo_radius function: http://halotools.readthedocs.org/en/master/api/halotools.empirical_models.halo_mass_to_halo_radius.html#halotools.empirical_models.halo_mass_to_halo_radius. 

All I was suggesting was that there be an exactly analogous feature called "Tested by". And any reference in such a section will appear in a specially marked box, just like "see also". In the same documentation for halo_mass_to_halo_radius, there is a single sentence in the "Notes" providing a link to a description of the test that the function is subject to. 

Other such links point to test docs that are more extensive. For example, http://halotools.readthedocs.org/en/master/api/halotools.empirical_models.NFWProfile.html#halotools.empirical_models.NFWProfile.cumulative_mass_PDF has test suite documentation link to http://halotools.readthedocs.org/en/master/api/halotools.empirical_models.phase_space_models.profile_models.tests.test_nfw_profile.TestNFWProfile.html#halotools.empirical_models.phase_space_models.profile_models.tests.test_nfw_profile.TestNFWProfile.test_cumulative_mass_PDF. 

I was thinking this could be accomplished even in a hacky, copy-and-paste way by cribbing from the "See also" sphinx code, but I have basically zero experience with this and so it is possible that this request is *far* more work than it is worth. If such a simple, cut-and-dried solution is possible, I do think this would help improve the clarity of the documentation. But it is perfectly acceptable in my view to just continue as-is. Again, similar to #273 , if there is an easy bang-for-the-buck solution then it's worth considering, but if it would require significant labor, then just say so and close the issue.

**Comment by aphearin** [comment]
I don't think this is worth spending time on. It's perfectly fine to just use the See also section for this purpose.
